api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -83,6 +84,19 @@ public QueueEntry saveQueueEntry(QueueEntry queueEntry) {
 				throw new IllegalArgumentException("Patient mismatch - visit.patient does not match patient");
 			}
 		}
+		QueueEntrySearchCriteria searchCriteria = new QueueEntrySearchCriteria();
+		searchCriteria.setPatient(queueEntry.getPatient());
+		searchCriteria.setQueues(Collections.singletonList(queueEntry.getQueue()));
+		List<QueueEntry> queueEntries = getQueueEntries(searchCriteria);
+		for (QueueEntry entry : queueEntries) {
```

Comment:
I feel like we would be best served if the logic below was extracted out.  This could potentially be a method directly on QueueEntry (eg. `boolean isOverlapping(QueueEntry)`) or could be something we add to QueueUtils (eg. `static boolean datesOverlap(Date startDate1, Date endDate1, Date startDate2, Date endDate2)`.  Then, we can add a lot of unit tests around this method to ensure that it meets all of the right behavior, and then when we test the actual service, we can potentially need to cover fewer of the various use cases.